internal/api/mail.go

```diff
@@ -575,34 +525,71 @@ func (a *API) sendEmail(r *http.Request, tx *storage.Connection, u *models.User,
 			EmailActionType: emailActionType,
 			RedirectTo:      referrerURL,
 			SiteURL:         externalURL.String(),
-			TokenHash:       tokenHashWithPrefix,
-		}
-		if emailActionType == mail.EmailChangeVerification && config.Mailer.SecureEmailChangeEnabled && u.GetEmail() != "" {
-			emailData.TokenNew = otpNew
-			emailData.TokenHashNew = u.EmailChangeTokenCurrent
+			TokenHash:       ott.TokenHash,
 		}
 		input := hooks.SendEmailInput{
 			User:      u,
 			EmailData: emailData,
 		}
 		output := hooks.SendEmailOutput{}
-		return a.invokeHook(tx, r, &input, &output, a.config.Hook.SendEmail.URI)
+		return a.invokeHook(tx, r, &input, &output, config.Hook.SendEmail.URI)
 	}
 
 	switch emailActionType {
 	case mail.SignupVerification:
-		return mailer.ConfirmationMail(r, u, otp, referrerURL, externalURL)
+		return mailer.ConfirmationMail(r, u, ott, otp, referrerURL, externalURL)
 	case mail.MagicLinkVerification:
-		return mailer.MagicLinkMail(r, u, otp, referrerURL, externalURL)
+		return mailer.MagicLinkMail(r, u, ott, otp, referrerURL, externalURL)
 	case mail.ReauthenticationVerification:
 		return mailer.ReauthenticateMail(r, u, otp)
 	case mail.RecoveryVerification:
-		return mailer.RecoveryMail(r, u, otp, referrerURL, externalURL)
+		return mailer.RecoveryMail(r, u, ott, otp, referrerURL, externalURL)
 	case mail.InviteVerification:
-		return mailer.InviteMail(r, u, otp, referrerURL, externalURL)
-	case mail.EmailChangeVerification:
-		return mailer.EmailChangeMail(r, u, otpNew, otp, referrerURL, externalURL)
```

Comment:
moved this into it's own function (see `sendEmailChangeEmails` because we're passing way too many arguments into the function just to satisfy this case

if we were to combine them, the `sendEmail` function would need to take in:
* `r *http.Request`
* `tx *storage.Connection` 
* `u *models.User`
* `emailActionType` 
* `otp, otpNew string`
* `ott, ottNew *models.OneTimeToken`

where `otpNew` and `ottNew` are only needed in the email change case.

further refactoring can be done in a subsequent PR to use the `ott.TokenType` rather than passing in an `emailActionType`